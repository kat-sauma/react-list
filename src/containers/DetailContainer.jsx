import React, { Component } from 'react'
import CharacterDetail from '../components/characters/CharacterDetail';
import { findCharacterById } from '../services/rmApiCalls';

export default class DetailContainer extends Component {
    state = {
        character: { }
    }

    async componentDidMount() {
        const { id } = this.props.match.params;

        const character = await findCharacterById(id);

        this.setState({
            character
        });
    }
    render() {
        console.log(this.state.character, 'hi');
        const { character } = this.state;
        return <CharacterDetail character={character} />;
    }
}


