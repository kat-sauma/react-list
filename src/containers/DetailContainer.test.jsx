import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailContainer from './DetailContainer';
import { MemoryRouter } from 'react-router-dom';

describe('DetailContainer', () => {
    it('renders details of character to the page', async () => {
        render(<MemoryRouter><DetailContainer match={{ 
            params: { 
                id: 1
            }
        }}/></MemoryRouter>);

        const li = await screen.findByRole('list', { name: 'character' });

        expect(li).toMatchSnapshot();
    });
});