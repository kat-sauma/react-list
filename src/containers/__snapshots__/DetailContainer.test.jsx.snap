// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContainer renders details of character to the page 1`] = `
<li
  aria-label="character page"
  class="character-page"
>
  <img />
  <p />
  <p />
  <p />
  <p />
</li>
`;
